File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0 ; i < nums.length ; i++) {
            int opposite = target - nums[i];

            if (map.containsKey(opposite)) {
                return new int[]{ map.get(opposite), i };
            }
            map.put(nums[i], i);
        }

        return new int[] {};
    }
}

File: Data Structures & Algorithms/search-in-rotated-sorted-array-ii/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {boolean}
     */
    search(nums: number[], target: number): boolean {
        for (const num of nums) {
            if (num === target) return true
        }
        return false
    }
}
